ipld/resolver/src/behaviour/mod.rs

```diff
@@ -29,22 +34,65 @@ impl NetworkConfig {
     }
 }
 
-/// Libp2p behaviour to manage content resolution from other subnets, using:
+#[derive(thiserror::Error, Debug)]
+pub enum ConfigError {
+    #[error("Error in the discovery configuration")]
+    Discovery(#[from] discovery::ConfigError),
+    #[error("Error in the membership configuration")]
+    Membership(#[from] membership::ConfigError),
+}
+
+/// Libp2p behaviour bundle to manage content resolution from other subnets, using:
 ///
 /// * Kademlia for peer discovery
 /// * Gossipsub to advertise subnet membership
 /// * Bitswap to resolve CIDs
 #[derive(NetworkBehaviour)]
-pub struct IpldResolver<P: StoreParams> {
+pub struct Behaviour<P: StoreParams> {
     ping: ping::Behaviour,
     identify: identify::Behaviour,
     discovery: discovery::Behaviour,
     membership: membership::Behaviour,
-    bitswap: content::Behaviour<P>,
+    content: content::Behaviour<P>,
 }
 
 // Unfortunately by using `#[derive(NetworkBehaviour)]` we cannot easily inspects events
 // from the inner behaviours, e.g. we cannot poll a behaviour and if it returns something
-// of interest then call a method on another behaviour. We can do this in another wrapper
+// of interest then call a method on another behaviour. We can do this in yet another wrapper
 // where we manually implement `NetworkBehaviour`, or the outer service where we drive the
-// Swarm; there we are free to call any of the behaviours.
+// Swarm; there we are free to call any of the behaviours as well as the Swarm.
+
+impl<P: StoreParams> Behaviour<P> {
+    pub fn new<S>(
+        nc: NetworkConfig,
+        dc: DiscoveryConfig,
+        mc: MembershipConfig,
+        store: S,
+    ) -> Result<Self, ConfigError>
+    where
+        S: BitswapStore<Params = P>,
+    {
+        Ok(Self {
+            ping: Default::default(),
+            identify: identify::Behaviour::new(identify::Config::new(
+                "ipfs/0.1.0".into(),
```

Comment:
I don't know, I copied it as-is from Forest. The `KademliaConfig` says the protocol name can be used to distinguish different groups of peers, but the `IdentifyConfig` just says:
```
    /// Application-specific version of the protocol family used by the peer,
    /// e.g. `ipfs/1.0.0` or `polkadot/1.0.0`.
    pub protocol_version: String,
```
I fixed the `0.1.0` to be `1.0.0`, it looked like a typo. I suspect it doesn't matter what it is, it could be `ipc`? I can't see it used, but I only looked at the `Identify` behaviour.

For the record when my nodes connect to each other, I see these protocol being supported:
```
["/ipfs/ping/1.0.0", "/ipfs/id/1.0.0", "/ipfs/id/push/1.0.0", "/ipc/smoke-test/kad/1.0.0", "/meshsub/1.1.0", "/meshsub/1.0.0", "/ipfs-embed/bitswap/1.0.0"]
```

So this exact thing doesn't even appear :shrug:

Reply:
I see they are using it for authentication purposes.